app.js

```diff
@@ -26,10 +66,17 @@ app.use(function(req, res, next) {
 });
 
 app.get('/validate', function (req, res, next) {
-  const token = firebase.auth().createCustomToken(req.query.token);
-  res.json({ token });
+  var apiToken = req.query.token;
+  if (isValidToken(apiToken)) {
+    getApiUser(apiToken).then(function() {
+      const token = firebase.auth().createCustomToken(req.query.uid);
+      res.json({ token });
+    })
+  } else {
+    console.log('Invalid token')
```

Comment:
e.g. [here](https://github.com/zooniverse/static-crop/blob/f01cca3727aad253c16ae3ea95f75337e3c64931/server.js#L105) and other places in that link we're providing some info about what went wrong in the req if error and if the params are invalid, etc.